spec.bs

```diff
@@ -491,12 +494,243 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
 
 A {{URLPattern}} is associated with multiple <dfn>component</dfn> [=structs=].
 
-A [=component=] has an associated <dfn for=component>pattern string</dfn>, a string, initially null.
+A [=component=] has an associated <dfn for=component>pattern string</dfn>, a [=/pattern string=], initially null.
 
 A [=component=] has an associated <dfn for=component>regular expression</dfn>, a {{RegExp}}, initially null.
 
 A [=component=] has an associated <dfn for=component>group name list</dfn>, an array of strings, initially an empty array.
 
+<h2 id=patterns>Patterns</h2>
+
+A <dfn>pattern string</dfn> is a string conforming to a particular syntax.  This pattern syntax is directly based on the syntax used by the popular [path-to-regexp](https://github.com/pillarjs/path-to-regexp) JavaScript library.
```

Comment:
Is this definition accurate? In particular it seems like the tokenizers use of pattern strings as inputs does not assume they conform to any particular syntax; it tokenizes anything you give it.

Maybe something like

> A pattern string is a string that will be processed as part of URL pattern tokenizing and parsing

?

Reply:
Hmm.  I'll rephrase to say that a "well formed" or "valid" pattern string conforms to a particular syntax.  And I can change the input of the tokenizer to not be a `[=pattern string=]`.

Reply:
Rephrased, although, I still not 100% sure about this yet.  Might need to redo it once the rest is written.

Reply:
It looks pretty good for now. Ideally you'd have a rigorous definition of well-formed that can be used by validators, but that's probably overkill. Maybe "well-formed" really means "already validated by the parser/tokenizer" or something in your case?

Anyway, yeah, leaving the exact distinction here for a polish phase makes sense.